app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
+      return [];
+    }
+    // find all existing experience domains
+    let sharedTableEntries: Array<MultipleUserTableEntry> = [];
+    users.forEach(user => {
+      Object.keys(user.experiences).forEach(experience => {
+        if (!sharedTableEntries.find(entry => entry.domain === experience)) {
+          sharedTableEntries.push({
+            domain: experience,
+            value: 0,
+            lowestValue: 0,
+            highestValue: -1,
+            isShared: true,
           });
-
-          return this.sendUserToServer(newUser, user);
         }
-        return Promise.resolve(user);
       });
-
-      this.closeModal(newUserPromises);
-    }
+    });
+    // create sharedTableEntries
+    sharedTableEntries = sharedTableEntries.map(entry => {
+      let highestValue = -1;
+      let lowestValue = -1;
+      let isShared = true;
+      const domain = entry.domain;
+      users.forEach(user => {
+        const experiences = user.experiences;
+        if (domain in experiences) {
+          highestValue = highestValue > experiences[domain] ? highestValue : experiences[domain];
+          lowestValue =
+            lowestValue < experiences[domain] && lowestValue !== -1
+              ? lowestValue
+              : experiences[domain];
+        } else {
+          isShared = false;
+          lowestValue = 0;
+        }
+      });
+      return { domain, highestValue, lowestValue, value: lowestValue, isShared };
+    });
+    // sort entries
+    return sharedTableEntries.sort(
+      Utils.localeCompareBy(([]: Array<MultipleUserTableEntry>), entry =>
+        entry.domain.toLowerCase(),
+      ),
+    );
   };
 
-  deleteExperience = () => {
-    if (this.state.domain) {
-      const { domain } = this.state;
-      const newUserPromises = this.props.users.map(user => {
-        if (this.props.selectedUserIds.includes(user.id)) {
-          const newExperiences = _.omit(user.experiences, domain);
-          const newUser = update(user, {
-            experiences: { $set: newExperiences },
-          });
+  loadSingleUserEntries = (users: Array<APIUserType>): Array<SingleUserTableEntry> =>
+    users.length === 1
+      ? _.map(users[0].experiences, (value, domain) => ({
+          domain,
+          value,
+        })).sort(
+          Utils.localeCompareBy(([]: Array<SingleUserTableEntry>), entry =>
+            entry.domain.toLowerCase(),
+          ),
+        )
+      : [];
 
-          return this.sendUserToServer(newUser, user);
-        }
-        return Promise.resolve(user);
+  updatedAllUsers = async () => {
+    let newUserPromises: Array<Promise<APIUserType>>;
+    if (this.state.showOnlySharedExperiences && this.props.selectedUsers.length > 1) {
+      const relevantEntries = this.state.multipleUsersEntries.filter(entry => entry.isShared);
+      newUserPromises = this.props.selectedUsers.map(user => {
+        const newExperiences = { ...user.experiences };
+        relevantEntries.forEach(entry => {
```

Comment:
Here, you are initializing `newExperiences` and then you update it in a `forEach` call. In general, it's nicer if you can define a variable directly with the right values so that you don't need any mutation. In this case, you can do something like this:

```
const newExperiences = { 
	...user.experiences,
	..._.fromPairs(relevantEntries.map({domain, value} => [domain, value]))
};
```

`_.fromPairs` takes something like `[["a", "b"], ["c", "d"]]` and returns `{a: "b", c: "d"}`. With the `...` spread, the new experiences are directly merged into the existing ones.